lib/manageiq/providers/ibm_cloud/cloud_tools/activity_tracker.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module ManageIQ
+  module Providers
+    module IbmCloud
+      module CloudTools
+        class ActivityTracker < Sdk::Branch
+          def initialize(cloudtools:, region:)
+            super(:cloudtools => cloudtools)
+            service_url = "https://api.#{region}.logging.cloud.ibm.com"
+            @sdk_params = {:service_url => service_url}
+          end
+
+          # Return a new Activty Tracker SDK instance.
+          def sdk_client
+            @sdk_params[:authenticator] = @cloudtools.authenticator
+            service_key = AuthUseridPassword.find_by(:authtype => "events").auth_key
```

Comment:
This is to access the record which was created in my local env to access the service key. The record was created using:
```
vpc = ManageIQ::Providers::IbmCloud::VPC::CloudManager.first
vpc.authentications.create!(:authtype => "events", :auth_key => <service_key>)
```

Reply:
So the problem with doing this here is that `AuthUseridPassword.find_by()` would look through the whole "authentications" table not just for this provider type or even this provider.